tools/build_langserver/langserver/handler.go

```diff
@@ -70,6 +71,18 @@ func (h *LsHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrp
 		req.Method != "initialzed" && req.Method != "shutdown" {
 		ctx = h.requestStore.Store(ctx, req)
 		defer h.requestStore.Cancel(req.ID)
+
+		go func() {
+			t, err := h.diagPublisher.queue.Get(1)
+			if err != nil {
+				log.Warning("fail to get diagnostic publishing task")
+			}
+			if len(t) > 0 {
+				task := t[0].(taskDef)
+
+				h.publishDiagnostics(conn, task.content, task.uri)
```

Comment:
do you not need a loop here? basically loop forever, consuming off the queue and processing them?